flake.nix

```
@@ -62,12 +150,31 @@
         channels:
         let
           pkgs = channels.nixpkgs;
+          lib = inputs.nixpkgs.lib;
+          instance = (instantiate_lib lib pkgs);
         in
         {
-          packages = callYaziPlugins pkgs;
           formatter = pkgs.nixfmt-rfc-style;
+          packages = instance.packages // {
+            homeManagerModules = rec {
+              yaziPlugins =
+                { lib, ... }:
+                {
+                  imports =
+                    ((instantiate_lib lib (inputs.nixpkgs.legacyPackages.x86_64-linux)).homeManagerModulesImports)
+                    ++ [ ./module.nix ];
+                };
+              default = yaziPlugins;
+            };
+          };
+
         };
 
-      overlays.default = final: prev: { yaziPlugins = callYaziPlugins prev; };
+      overlays.default =
+        let
+          lib = inputs.nixpkgs.lib;
+        in
+        final: prev: { yaziPlugins = (instantiate_lib lib prev).packages; };
+
```

This will use the `lib` from our flake's input, but maybe it'd be better to use the `lib` from whatever nixpkgs the overlay is applied to? I think the following would work, but haven't tried to eval it in context so there might be infinite recursion due to `final`:
```suggestion
      overlays.default = final: prev: { yaziPlugins = (instantiate_lib (final.lib) prev).packages; };

```